### PR TITLE
feat: allow expiration date for .trivyignore entries

### LINKED ISSUE #1450: Consider adding deferring of encountered issues to existing .trivyignore file
Currently, the .trivyignore file allows you to accept risks, but most of the time you don't want to ignore a risk, you want to defer it, since either you can't do it yet, or there is no known fix for an issue currently.

In these cases, it would be helpful to apply a timestamp to the ignore file after which the vulnerability would be reported again (indicating that it was deferred).

Example implementation (adapted from GITHUB example):

``` bash
$ cat .trivyignore
# Accept the risk (till the new year, there's partying to be had)
CVE-2018-14618 2022-01-01T00:00:00+0000
```

### PR DESCRIPTION
**Pull request by maltemorgenstern**
## Description
Allow users to specify an expiration date for `.trivyignore` entries. This way a CVE can be ignored for a specific amount of time but it will be reported again afterwards.

```
$ cat .trivyignore
# Accept the risk
CVE-2018-14618

# Accept the risk until 2022-09-01
CVE-2019-14697 exp:2022-09-01
```

Expiration dates partially follow [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339) layout but do not define hours/minutes/seconds - just dates.

## Related issues
- Close #1450

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
Thanks for your contribution! Could you resolve conflicts?

**Comment by maltemorgenstern** [comment]
> Thanks for your contribution! Could you resolve conflicts?

Done 😃

File: pkg/result/filter.go [hunk 1]
```diff
@@ -210,14 +211,39 @@ func getIgnoredIDs(ignoreFile string) []string {
 		if strings.HasPrefix(line, "#") || line == "" {
 			continue
 		}
-		ignoredIDs = append(ignoredIDs, line)
+		// Process all fields
+		fields := strings.Fields(line)
+		if len(fields) > 1 {
+			exp, err := getExpirationDate(fields)
+			if err != nil {
+				log.Logger.Warnf("Error while parsing expiration date in .trivyignore file: %s\n", err)
```
Comment (knqyf263) [inline pkg/result/filter.go:219]: nit: you don't need `\n`.

```suggestion
				log.Logger.Warnf("Error while parsing expiration date in .trivyignore file: %s", err)
```

File: pkg/result/filter.go [hunk 2]
```diff
@@ -210,14 +211,39 @@ func getIgnoredIDs(ignoreFile string) []string {
...
 
 	log.Logger.Debugf("These IDs will be ignored: %q", ignoredIDs)
 
 	return ignoredIDs
 }
 
+func getExpirationDate(fields []string) (time.Time, error) {
+	for _, field := range fields {
+		if strings.HasPrefix(field, "exp:") {
+			return time.Parse(time.RFC3339, strings.TrimPrefix(field, "exp:"))
```
Comment (knqyf263) [inline pkg/result/filter.go:240]: I guess most users don't want to specify hour/minute/second. What if we define a layout like `"2006-01-02"`?

Comment (maltemorgenstern) [reply pkg/result/filter.go]: Yeah, that absolutely makes sense 👍 
I changed it to `2006-01-02` and updated the tests, docs and PR description.

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
Thanks a lot!

**Comment by maltemorgenstern** [comment]
> Thanks a lot!

No problem - looking forward to using this in our projects aswell 🎉

**Pull request merged**